### DISCUSSION TITLE
Oh My Posh error in Cygwin (mintty + zsh)

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [JanDeDobbeleer/oh-my-posh#1799] [Troubleshoot]
I've just started using Cygwin, and I've ran into an error. I'm using my regular zsh config (from Git Bash + zsh), and i modified my cygwin config so `~` evaluates to my proper home folder.
Setting the shell to `pwsh` or `bash` did not help.

```zsh
eval "$(oh-my-posh --init --shell bash --config ~/theonlytails.omp.json)"
```

**Comment by JanDeDobbeleer** [comment]
What error do you get?

**Reply by TheOnlyTails** [reply]
Literally just
```
Oh My Post Error:
```

**Comment by JanDeDobbeleer** [comment]
@TheOnlyTails can you share you exact setup? What I have to work with:

- zsh on Windows
- Git bash

For zsh, you do need to set `--shell zsh`.

Both work as expected on my end.

**Reply by TheOnlyTails** [reply]
I installed zsh and mintty with Cygwin, and then created a Windows Terminal profile to launch zsh from mintty.

**Reply by JanDeDobbeleer** [reply]
Why would you do that when you can just install zsh natively? Because I feel all sorts of things can go wrong in that case, Cygwin is already a monster of a utility to work with, let alone in a setup like this. I set it up natively following [this](https://gist.github.com/fworks/af4c896c9de47d827d4caa6fd7154b6b) guide and everything works. Anything else is discouraged imho.

**Reply by TheOnlyTails** [reply]
it's mostly about trouble with git bash's support for ANSI colors in some cases. I do see your point though, thanks.

**Reply by JanDeDobbeleer** [reply]
I can confirm that installation method makes everything work. I can share my Windows Terminal setup too if you'd like.

**Reply by TheOnlyTails** [reply]
I've mostly had trouble with tools like https://github.com/extrawurst/gitui. oh my posh is working fine.
